packages/back-end/src/util/scheduleRules.ts

```diff
@@ -0,0 +1,46 @@
+import { ScheduleRule } from "../../types/feature";
+
+export const getCurrentEnabledState = (
+  scheduleRules: ScheduleRule[],
+  currentDate: Date
+) => {
+  // If a feature doesn't have any rules, don't filter it out
+  if (!scheduleRules.length) {
+    return true;
+  }
+
+  // Ensure the scheduleRules array is always sorted earliest to latest
+  const sortedRules = scheduleRules.sort((a, b) => {
```

Comment:
@jdorn @bryce-fitzsimons I moved this to a util function and also added some logic to sort the `scheduleRules` by date, which requires some colorful logic.

Some background:
- Currently, a `scheduleRules` array can only have a length of 2, or be null, we don't support an array with a length of 1. 
- We do have some protections on the front end so the end date can't be before the start date, but I'm sure there are gotchas there, so I like the idea of sorting on the backend.
- This was built with a future in mind that would allow more than just a start/end date. For example, a user could start a feature immediately, disable it on 12/30/22, and then re-enable it on 1/05/23. The current logic does support that.

I've outlined what potential scheduleRule arrays could look like below.
```
// This is a rule that will launch immediately after being published, and turn off on 12/30/22.
[
{enableFeature: true, timestamp: null},
{enableFeature: false, timestamp: "2022-12-30T13:00:00.000Z"},
]

// This is a rule that will turn on, on 12/01/22 and will only turn off when the org manually disables it.
[
{enableFeature: true, timestamp: "2022-12-30T13:00:00.000Z"},
{enableFeature: false, timestamp: null },
]

// This is what a scheduleRule array looks like if the org wants it to start on 12/01/22 and turn off on 12/30/22

[
{enableFeature: true, timestamp: "2022-12-01T13:00:00.000Z"},
{enableFeature: false, timestamp: "2022-12-30T13:00:00.000Z" },
]
```

The colorful logic comes when we attempt to sort the array by date, when we do the date compare, we can't compare the date to `null`. So currently, the sort logic says, if a date is null, and the null date's corresponding `enableFeature` is true, that means this is a start rule, so I picked a really early date. But, if the date is null, and the date's corresponding `enableFeature` is false, that means it's an end date, so I picked a date in the far future.

While this works, it feels hacky. Have you come across a pattern you like to handle this sort of thing before?